src/main/kotlin/com/tschuchort/compiletesting/KaptComponentRegistrar.kt

```diff
@@ -40,32 +40,30 @@ import org.jetbrains.kotlin.kapt3.AbstractKapt3Extension
 import org.jetbrains.kotlin.kapt3.Kapt3ComponentRegistrar
 import org.jetbrains.kotlin.kapt3.base.Kapt
 import org.jetbrains.kotlin.kapt3.base.LoadedProcessors
-import org.jetbrains.kotlin.kapt3.base.incremental.IncrementalProcessor
 import org.jetbrains.kotlin.kapt3.base.util.KaptLogger
 import org.jetbrains.kotlin.kapt3.util.MessageCollectorBackedKaptLogger
 import org.jetbrains.kotlin.psi.KtFile
 import org.jetbrains.kotlin.resolve.BindingTrace
 import org.jetbrains.kotlin.resolve.jvm.extensions.AnalysisHandlerExtension
 import java.io.File
-import javax.annotation.processing.Processor
-
-internal class KaptComponentRegistrar : ComponentRegistrar {
 
+class KaptComponentRegistrar: ComponentRegistrar {
     override fun registerProjectComponents(project: MockProject, configuration: CompilerConfiguration) {
-        if (threadLocalParameters.get().processors.isEmpty())
+        if (CommonComponentRegistrar.threadLocalParameters.get().processors.isEmpty())
```

Comment:
Why not just pass those as constructor parameters, now that we instantiate the `KaptComponentRegistrar` manually?